students_proj/src/main/java/com/musala/simple/students/Main.java

```diff
@@ -0,0 +1,172 @@
+package com.musala.simple.students;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+import com.google.gson.Gson;
+
+public class Main {
+
+	public static void main(String[] args) {
+
+		String path;
+		
+		// Check if any arguments are provided
+		if (args.length == 0) {
+			throw new IllegalArgumentException("Path is missing.");
+		}
+		
+		path = args[0];
+    	File file = new File(path);
+
+    	// Check if path exists
+    	if (!file.exists()) {
+    		throw new IllegalArgumentException("Invalid path specified.");
+    	}
+    		
+    	// Check if the valid path is a file
+	    if (!file.isFile()) {
+	    	throw new IllegalArgumentException("The path specified is not an existing file.");
+	    }	
+	    
+        // Check the file extension to be JSON.
+	    boolean isJson = path.endsWith(".json");
+        if (!isJson) {
+        	throw new IllegalArgumentException("The file specified is not of '.json' format.");
+        } 
+      
+        String fileContent = readFile(path);
+        int requestedStudentId = readStudentId(args);
+        
+        // validate JSON
+        if (!isValidJson(fileContent)) {
+        	throw new IllegalArgumentException("The file given does not contain a valid JSON");
+        }
+        
+		Gson gson = new Gson();
+		Wrapper wrapper = gson.fromJson(fileContent, Wrapper.class);
+    	StudentGroup studentGroup = new StudentGroup();
+    	studentGroup.fillStudentGroup(wrapper.students);
+
+    	List<Student> students = studentGroup.getStudents();	
+    	
+    	boolean studentExists = false;
+    	if (requestedStudentId != -1) {
+    		Student student;
+    		try {
+    			student = studentGroup.getStudentById(requestedStudentId);
+    			studentExists = true;
+    			printStudentDetails(student);
+    		} catch (IllegalArgumentException iae) {
+    			System.err.println("Requested student does not exist in the data.");
+    		} 
+    	} 
+    	if (!studentExists) {
+    		printStudents(students);
+    	}
+	}
+	
+	/**
+	 * Tries to parse a String using the {@link Gson#fromJson(com.google.gson.JsonElement, Class)
+	 * method, which throws an Exception if the parameter String is
+	 * not a valid Json.
+	 *
+	 * @param  Json  A String that the user wants to validate as JSON
+	 * @return validation of weather the given String is a valid JSON
+	 */
+	private static boolean isValidJson (String Json) {
+        Gson gson = new Gson();
+        try {
+            gson.fromJson(Json, Object.class);
+            return true;
+        } catch (com.google.gson.JsonSyntaxException ex) {
```

Comment:
Import `com.google.gson.JsonSyntaxException` so that the statement becomes `catch (JsonSyntaxException ex)`.